// stimulus-timeago/dist/stimulus-timeago.mjs
import { Controller as s } from "@hotwired/stimulus";
import { formatDistanceToNow as a } from "date-fns";
class r extends s {
  initialize() {
    this.isValid = !0;
  }
  connect() {
    this.load(), this.hasRefreshIntervalValue && this.isValid && this.startRefreshing();
  }
  disconnect() {
    this.stopRefreshing();
  }
  load() {
    const e = this.datetimeValue, i = Date.parse(e), t = {
      includeSeconds: this.hasIncludeSecondsValue,
      addSuffix: this.hasAddSuffixValue,
      locale: this.locale
    };
    Number.isNaN(i) && (this.isValid = !1, console.error(
      `[stimulus-timeago] Value given in 'data-timeago-datetime' is not a valid date (${e}). Please provide a ISO 8601 compatible datetime string. Displaying given value instead.`
    )), this.element.dateTime = e, this.element.innerHTML = this.isValid ? a(i, t) : e;
  }
  startRefreshing() {
    this.refreshTimer = setInterval(() => {
      this.load();
    }, this.refreshIntervalValue);
  }
  stopRefreshing() {
    this.refreshTimer && clearInterval(this.refreshTimer);
  }
}
r.values = {
  datetime: String,
  refreshInterval: Number,
  includeSeconds: Boolean,
  addSuffix: Boolean
};
export {
  r as default
};
